### DISCUSSION TITLE
Easier method for scripture Input

### DISCUSSION THREAD

**Discussion by genu** [Joomla-Bible-Study/Proclaim#846] [Ideas]
For the longest time, the scripture has been inputed as: book, chapter_begin, verse_begin, verse_end, chapter_end, book2, chapter_begin2 etc.

A good improvement to this, is to allow the user to enter scripture references directly like so:

`John 3:16, Romans 1:16`

I'm thinking this could be implemented as a single text field with autocomplete, so that only correct scripture references are acceptable.

A utility class will need to be created to validate a given input to test whether its a valid scripture or not. (i.e. `John 3:16` would be valid, while `john 0:16` would be invalid). The user is alerted whether a references they entered is valid or not as they're entering it. Multiple references are simply separated by a comma.

Taking this approach, will allow us to truncate more than 10 columns in the message database.
#### Related Tasks
- It will require a data migration from the old format.

**Comment by tomfuller2** [comment]
Great idea, Eugen. I originally modeled the Scripture entry on SermonCentral.com.

**Comment by bcordis** [comment]
We still cold us the tables as is but just do some JS to fill in the right
fields. We still need the each filed I think.

On Tue, Apr 9, 2013 at 9:52 AM, tomfuller2 notifications@github.com wrote:

> Great idea, Eugen. I originally modeled the Scripture entry on
> SermonCentral.com.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Joomla-Bible-Study/Joomla-Bible-Study/issues/147#issuecomment-16117229
> .

## 

Thanks,
Brent Cordis
2406 Hermitage Park Dr.
Hermitage, TN 37076
615.429.6426

**Comment by genu** [comment]
@bcordis 
The challenge would probably be to update all the references to the message scripture references, and filters.

The filters by default, I believe, work only on column values, so for filtering by a specific book under this method is a little bit more complex.

If we implement this in a utility class (or library?), then the burden would be on this utility class of providing proper methods for handling filtering, localization, and other required functionality.

**Comment by bcordis** [comment]
@genu 
Like in contacts there is the ability to explode out the entry into a object db and then same each one into there appropriate table columns?

**Comment by genu** [comment]
I didn't look how the contacts component works.

We could, as a first step in this direction, take that route and just map a string of scripture references to those table columns. However, we will still be limited by only 2 scripture references per message. correct?

**Comment by bcordis** [comment]
unless we redo the db and do it in a JSON string in one column.

On Tue, Apr 9, 2013 at 11:06 AM, Eugen Istoc notifications@github.comwrote:

> I didn't look how the contacts component works.
> 
> We could, as a first step in this direction, take that route and just map
> a string of scripture references to those table columns. However, we will
> still be limited by only 2 scripture references per message. correct?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Joomla-Bible-Study/Joomla-Bible-Study/issues/147#issuecomment-16122506
> .

## 

Thanks,
Brent Cordis
2406 Hermitage Park Dr.
Hermitage, TN 37076
615.429.6426

**Comment by bcordis** [comment]
This has been fully implemented in PR #1056 (merged into `development`).

**What was built:**
- Single autocomplete text input for natural references like "Luke 7:36-38" with progressive hints and book name suggestions
- New `#__bsms_study_scriptures` junction table supports unlimited scripture references per message (no more 2-slot limit)
- Per-reference Bible version via searchable dropdown
- Drag-to-reorder, dark theme support
- Layout Editor "All Scriptures" element replaces deprecated scripture1/scripture2
- Configurable separator, optional passage headings, Display Defaults tab
- Migration script syncs first 2 junction refs back to legacy columns for backward compatibility
- Finder plugin indexes all junction table scriptures
- mod_proclaim loads junction table scriptures for identical rendering

This addresses the original request from 2013 — scripture entry is now a single text field with autocomplete, unlimited references, and proper validation. The old 10-column approach is fully replaced.
